pychunkedgraph/io/edges.py

```diff
@@ -36,38 +36,38 @@ def deserialize(edges_message: EdgesMsg) -> Tuple[np.ndarray, np.ndarray, np.nda
     return Edges(sv_ids1, sv_ids2, affinities=affinities, areas=areas)
 
 
-def _decompress_edges(content: bytes) -> Dict:
+def _parse_edges(compressed: List[bytes]) -> List[Dict]:
     zdc = zstd.ZstdDecompressor()
-    chunk_edges = ChunkEdgesMsg()
-    chunk_edges.ParseFromString(zdc.decompressobj().decompress(content))
-
-    # in, between and cross
-    edges_dict = {}
-    edges_dict[EDGE_TYPES.in_chunk] = deserialize(chunk_edges.in_chunk)
-    edges_dict[EDGE_TYPES.between_chunk] = deserialize(chunk_edges.between_chunk)
-    edges_dict[EDGE_TYPES.cross_chunk] = deserialize(chunk_edges.cross_chunk)
-    return edges_dict
+    decompressed = zdc.multi_decompress_to_buffer(compressed, threads=4)
+    result = []
+    for content in decompressed:
+        content = content.tobytes()
+        chunk_edges = ChunkEdgesMsg()
+        chunk_edges.ParseFromString(content)
+        edges_dict = {}
+        edges_dict[EDGE_TYPES.in_chunk] = deserialize(chunk_edges.in_chunk)
+        edges_dict[EDGE_TYPES.between_chunk] = deserialize(chunk_edges.between_chunk)
+        edges_dict[EDGE_TYPES.cross_chunk] = deserialize(chunk_edges.cross_chunk)
+        result.append(edges_dict)
+    return result
 
 
 def get_chunk_edges(edges_dir: str, chunks_coordinates: List[np.ndarray]) -> Dict:
     """Read edges from GCS."""
-    from cloudfiles import CloudFiles
-
     fnames = []
     for chunk_coords in chunks_coordinates:
         chunk_str = "_".join(str(coord) for coord in chunk_coords)
         # filename format - edges_x_y_z.serialization.compression
         fnames.append(f"edges_{chunk_str}.proto.zst")
 
-    cf = CloudFiles(edges_dir)
+    cf = CloudFiles(edges_dir, num_threads=4)
```

Comment:
This is not exactly about multi-thread decompression so kind of off-topic, but I do not mind to have it since this speeds things up.